referencing resources from repositories with submodules yield runtime error

**What happened**:
```
panic: runtime error: index out of range [2] with length 1
goroutine 1 [running]:
github.com/gardener/docforge/pkg/resourcehandlers/github.TreeEntryToGitHubLocator(0xc000219880, 0xc0000943fc, 0x6, 0x0)
        /Users/i030268/go/src/github.com/gardener/docforge/pkg/resourcehandlers/github/github_resource_handler.go:60 +0x639
github.com/gardener/docforge/pkg/resourcehandlers/github.(*GitHub).URLToGitHubLocator(0xc000175ef0, 0x180ef40, 0xc0002e6000, 0xc0000943c0, 0x5f, 0xc0002a3901, 0x18145e0, 0x1, 0x3)
        /Users/i030268/go/src/github.com/gardener/docforge/pkg/resourcehandlers/github/github_resource_handler.go:365 +0x1e5
github.com/gardener/docforge/pkg/resourcehandlers/github.(*GitHub).ResolveDocumentation(0xc000175ef0, 0x180ef40, 0xc0002e6000, 0xc0000943c0, 0x5f, 0x51fffff, 0x203000, 0x203000)
        /Users/i030268/go/src/github.com/gardener/docforge/pkg/resourcehandlers/github/github_resource_handler.go:441 +0x65
github.com/gardener/docforge/pkg/reactor.(*Reactor).resolveNodeSelector(0xc0000bb920, 0x180ef40, 0xc0002e6000, 0xc00010c7e0, 0x0, 0xc000287640, 0x10, 0x10)
        /Users/i030268/go/src/github.com/gardener/docforge/pkg/reactor/reactor.go:215 +0xe4
github.com/gardener/docforge/pkg/reactor.(*Reactor).resolveStructure(0xc0000bb920, 0x180ef40, 0xc0002e6000, 0xc0002dc140, 0x4, 0x4, 0x0, 0x169da60, 0x1807701)
        /Users/i030268/go/src/github.com/gardener/docforge/pkg/reactor/reactor.go:190 +0x165
github.com/gardener/docforge/pkg/reactor.(*Reactor).ResolveManifest(0xc0000bb920, 0x180ef40, 0xc0002e6000, 0xc000175ad0, 0xc000287640, 0xc0002dc480)
        /Users/i030268/go/src/github.com/gardener/docforge/pkg/reactor/reactor.go:137 +0x137
github.com/gardener/docforge/pkg/reactor.(*Reactor).Run(0xc0000bb920, 0x180ef40, 0xc0002e6000, 0xc000175ad0, 0xc0000bb900, 0x0, 0x0)
        /Users/i030268/go/src/github.com/gardener/docforge/pkg/reactor/reactor.go:100 +0xef
github.com/gardener/docforge/cmd/app.NewCommand.func1(0xc0000e3080, 0xc00010c120, 0x0, 0x9, 0x0, 0x0)
        /Users/i030268/go/src/github.com/gardener/docforge/cmd/app/cmd.go:70 +0x1c9
github.com/spf13/cobra.(*Command).execute(0xc0000e3080, 0xc0000c0010, 0x9, 0x9, 0xc0000e3080, 0xc0000c0010)
        /Users/i030268/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000e3080, 0x1c089c8, 0x0, 0x0)
        /Users/i030268/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/i030268/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
main.main()
        /Users/i030268/go/src/github.com/gardener/docforge/cmd/main.go:43 +0x225
```

**What you expected to happen**:
Read resources successfully, skipping submodules.

**How to reproduce it (as minimally and precisely as possible)**:
Use a manifest that references any file (or features nodesSelector) in a repository with a submodule

**Anything else we need to know**:
From GitHub API perspective, a tree list object that is submodule will be of type `commit` with file mode `160000` and no URL (which is the root cause of the problem)

**How to categorize this issue?**
/kind bug
/priority normal

--- a/pkg/resourcehandlers/github/github_resource_handler.go
+++ b/pkg/resourcehandlers/github/github_resource_handler.go
@@ -30,7 +30,7 @@ var (
 )
 
 // TreeEntryToGitHubLocator creates a ResourceLocator from a github.TreeEntry and shaAlias.
-// The shaAlias is the name of e.g. a branch or a tag that should resolve to this resource
+// The shaAlias is a ref i.e. the name of e.g. a branch or a tag that should resolve to this resource
 // in the git database. It binds the formats of a GitHub website URLs to the GitHub API URLs.
 //
 // Example tree entries:
@@ -51,6 +51,11 @@ var (
 //	"url": "https://api.github.com/repos/gardener/gardener/git/blobs/91776959202ec10db883c5cfc05c51e78403f02c"
 //}
 func TreeEntryToGitHubLocator(treeEntry *github.TreeEntry, shaAlias string) *ResourceLocator {
+	// Tree entries such as (submodule) `commit` objects do not have URL
+	// and cannot be converted to ResourceLocator
+	if treeEntry.URL == nil {
+		return nil
+	}
 	url, err := url.Parse(treeEntry.GetURL())
 	if err != nil {
 		panic(fmt.Sprintf("failed to parse url %v: %v", treeEntry.GetURL(), err))

